hex/Cargo.toml

```diff
@@ -0,0 +1,33 @@
+[package]
+name = "fast-hex"
+version = "0.1.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>"]
+license = "CC0-1.0"
+description = "Fast hex encoding and decoding"
+homepage = "https://github.com/rust-bitcoin/hex/"
+repository = "https://github.com/rust-bitcoin/"
+documentation = "https://docs.rs/fast-hex/"
+keywords = ["encoding", "hex", "hexadecimal"]
+readme = "README.md"
+edition = "2018"
+
+[features]
+default = ["std"]
+std = ["alloc"]
+# If you disable std, you can still use Read/Write via the "io" feature.
+io = ["core2"]
```

Comment:
Why rename the feature? We shouldn't need it, just use `core2`.

Reply:
I thought for downstream users "io" might make it easier to understand whats going on
```
rust-bitcoin = { version = "..", default-features = false, features = ["alloc", "io"]
```
Is obvious with having to lookup what the "io" feature is, as opposed to "core2". Although it's maybe only clear because I wrote it, perhaps "io" does not mean `Read` and `Write` _only_ (which is how we are using `core2`)?

Reply:
Well, it goes against what is considered idiomatic. I don't find `io` particularly clear because it may suggest that the crate will be doing some I/O while it could be that it only uses in-memory operations. While `core2` is now required for some crates, we should really make our own traits and get rid of it.

Reply:
Ok, that's a strong enough argument for me. Will remove it.